### DISCUSSION TITLE
Failing to fetch data from tables in expo

### DISCUSSION THREAD

**Discussion by AviAvinav** [supabase/supabase#9827] [Questions]
I am creating a react native app with expo. I am using supabase for the first time and I wanted to fetch data from one of my tables, this is the function I am using to try it out:

```ts
const getTestChat = async () => {
    const { data } = await supabase
      .from("test")    // I tried changing values here from "test" to just random things but it didn't throw an error just the same result
      .select("*");

    return data;
  };

  console.log(getTestChat());
```

I created a table named test but all I get returned in the console is  `LOG  {"_1": 0, "_2": 0, "_3": null, "_4": null}`

This is the table in the table editor:

<img src="https://i.ibb.co/KsPBxf8/supabase.png" />

Can someone please help me with this?

**Comment by GaryAustin1** [comment]
Seems like you are calling an async function without an await or .then, so are getting a promise.   If you console log inside the function after the await you can verify if the Supabase part is working.

**Reply by AviAvinav** [reply]
So, I tried this:

```
const getTestChat = async () => {
    const { data } = await supabase.from("test").select("*");

    console.log(data);
  };
```

and then I called it with onPress in a TouchableOpacity and it returns this:

```
 LOG  []
```

What am I doing wrong?
